app/src/main/java/org/thoughtcrime/securesms/ui/theme/ThemeFromPreferences.kt

```diff
@@ -0,0 +1,65 @@
+package org.thoughtcrime.securesms.ui.theme
+
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.graphics.Color
+import org.session.libsession.utilities.TextSecurePreferences
+import org.session.libsession.utilities.TextSecurePreferences.Companion.BLUE_ACCENT
+import org.session.libsession.utilities.TextSecurePreferences.Companion.ORANGE_ACCENT
+import org.session.libsession.utilities.TextSecurePreferences.Companion.PINK_ACCENT
+import org.session.libsession.utilities.TextSecurePreferences.Companion.PURPLE_ACCENT
+import org.session.libsession.utilities.TextSecurePreferences.Companion.RED_ACCENT
+import org.session.libsession.utilities.TextSecurePreferences.Companion.YELLOW_ACCENT
+
+
+/**
+ * Returns the compose theme based on saved preferences
+ * Some behaviour is hardcoded to cater for legacy usage of people with themes already set
+ * But future themes will be picked and set directly from the "Appearance" screen
+ */
+@Composable
```

Comment:
This function is only `@Composable` because of the call to `isSystemInDarkTheme()`.

Perhaps extract everything else to a non-composable? or use `val lightDarkColors: StateFlow<LightDarkColors>` from [this PR](url)

Reply:
It is always called from Compose so there is no issue with it being Composable. You do need to know what the system is currently set to `isSystemInDarkTheme` in addition to the preferences chosen by the user.
It can be cleaned up once we have the sharedPref code if that manages to help.